Pegasus/include/Math.hpp

```diff
@@ -18,11 +18,87 @@
 #include <set>
 #include <unordered_map>
 #include <unordered_set>
+#include <numeric>
 
 namespace pegasus
 {
 namespace math
 {
+namespace fp
+{
+constexpr double g_floatingPointThreshold = 1e-4;
```

Comment:
I think at some point when we would have some actual data to work with, we would put it into some sort of a config file and allow the user to change it.